### ISSUE TITLE
a11y: Auto-set focus to be on the first option of a dropdown menu (iOS, Android, Mac)

### ISSUE THREAD

**Issue by pearlyong** [opengovsg/FormSG#5502]
**Problem**
When users encounter the combobox, it reads out as an edit box which is correct but is potentially confusing as a combobox is a cross between editbox and dropdown. When users Enter in editbox, the focus either jumps to the bottom of the page or does nothing, depending on the device.

**Related issue**
- On iOS Voiceover, when the dropdown opens if a user swipes it will lead them to the X button instead of going down the list

**Proposal**
- If users press Enter or Space, open the dropdown list and focus on the first item in the group.

**Comment by staceytan1998** [comment]
There is another issue regarding the dropdown,

This issue applies to Question 7: How would you rate the following experiences?

When the drop down menu for the ratings are expanded, the user would expect the values to be read in the following manner:
Strongly Agree, Agree, etc.
However, during the original test session, it was instead read as:
1 of 0, 2 of 0, etc.

This suggests that there is no accessible label for the drop down menu items.
However, testing with other Android devices running Android 12 and 13 using TalkBack, this issue did not occur.


but is **not replicable across different devices**, so just take note of the following: 

Ensure accessible labels for each drop down menu item are using ARIA, such as:
Strongly Agree, Agree, etc.

Because this issue is not replicable across different devices, it could be an issue of an out-of-date version of TalkBack being used during the time of testing.

**Comment by staceytan1998** [comment]
Update to above: Could not replicate the issue on Hui Qing's device

**Comment by staceytan1998** [comment]
**Issue**
When a user double taps to select the drop-down menu / combo box element to select an option, the element gets activated but the focus of the screen reader is not auto-set to land on the expanded drop-down menu options.

This makes it difficult to navigate and make a selection of choice since the user does not have the screen reader focus automatically set to the options, and neither can the user simply swipe to enter the options. The user has to manually move their finger around the screen randomly until they detect the presence of the expanded drop-down menu options.

Note 1: This issue occurs at all questions with the drop-down menu form field, i.e. Q3 and Q7.

Note 2: This issue is replicable on iOS (iPhone), iPad OS, and Android. This seems to be an underlying issue with how the focus indicator for screen readers on touchscreen devices interacts with the particular drop-down element. There are no problems using the drop-down menu question type on Windows using NVDA and Jaws screen readers.

Note 3: While testing on February 19, this issue did not exist on iOS during informal testing. However, follow-up testing on March 1 showed the issue existed on iOS and iPad OS.

**Suggestion**
Either of the following should be implemented:

1) Ensure the focus for screen readers will be auto-set to the options in the drop-down menu once it is selected and expanded.

2) An alternative is to use default HTML select tags since it comes with built-in accessibility features. However, the team understands this comes with design limitations and lacks the ability to allow the user to perform search for options.

However, from a UX point of view, it is important to consider if it is essential for the search functionality to be present for drop-down menus. This added complexity may not be easily understood for certain user groups such as senior citizens or persons with intellectual disabilities.

It might instead be prudent to encourage form administrators not to provide too many options in drop-down menus, and the development team might consider removing the search functionality to reduce the complexity of using the drop-down menus.

**Screenshot**
[Android](https://drive.google.com/file/d/1xrmXqX06RtqjNxXzhv0DzqHCyeeEvZ4B/view?usp=share_link)
[iOS](https://drive.google.com/file/d/1CqNpy__No8bXqeq9JRgkHc04nb34x-cw/view?usp=share_link)

**Comment by foochifa** [comment]
Reopening, as the issue happens on mobile which still persists

**Comment by foochifa** [comment]
Considering using a drawer for mobile users for the dropdown menu, waiting for design update on this!

**Comment by foochifa** [comment]
Design for reference https://www.figma.com/file/40SFxTL0j4xJTnU974t5v7/%5BOGP%5D-Base-Design-System?node-id=812-27503&t=anbFUWuob2a3WwjO-4
